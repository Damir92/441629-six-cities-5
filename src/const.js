export const PERCENT_PER_POINT_RATING = 20;

export const Routes = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_PAGE: `/offer/:id`,
  OFFER_LINK: `/offer`,
};

export const APIRoutes = {
  REVIEWS: `/comments`,
  OFFERS: `/hotels`,
  LOGIN: `/login`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const Cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const LivingType = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`,
};

export const Sorting = {
  POPULAR: `popular`,
  PRICE_TO_HIGH: `to-high`,
  PRICE_TO_LOW: `to-low`,
  TOP_RATED: `top-rated`,
};

export const SortingTitles = {
  [Sorting.POPULAR]: `Popular`,
  [Sorting.PRICE_TO_HIGH]: `Price: low to high`,
  [Sorting.PRICE_TO_LOW]: `Price: high to low`,
  [Sorting.TOP_RATED]: `Top rated first`,
};

export const SortingTypes = [
  Sorting.POPULAR,
  Sorting.PRICE_TO_HIGH,
  Sorting.PRICE_TO_LOW,
  Sorting.TOP_RATED,
];

export const MonthNames = {
  0: `January`,
  1: `February`,
  2: `March`,
  3: `April`,
  4: `May`,
  5: `June`,
  6: `July`,
  7: `August`,
  8: `September`,
  9: `November`,
  10: `October`,
  11: `December`,
};
