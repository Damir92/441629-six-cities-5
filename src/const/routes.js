const ROUTES = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_PAGE: `/offer/:id`,
};

export {ROUTES};
