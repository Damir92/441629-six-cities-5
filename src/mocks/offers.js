export const offers = [
  {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!',
    'goods': [
      'Fridge',
      'Washer',
      'Coffee machine',
      'Washing machine',
      'Dishwasher',
      'Laptop friendly workspace',
      'Cable TV',
      'Air conditioning',
      'Baby seat',
      'Towels',
      'Breakfast',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 2,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.37454,
      'longitude': 4.881976,
      'zoom': 16,
    },
    'volume': 1,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
    'price': 111,
    'rating': 3.3,
    'title': 'Wood and stone place',
    'type': 'room',
  },
  {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.',
    'goods': [ 'Laptop friendly workspace', 'Baby seat', 'Towels', 'Air conditioning', 'Breakfast', 'Washer' ],
    'host': { 'avatar': 'img/avatar-angelina.jpg', 'id': 25, 'isPro': true, 'name': 'Angelina' },
    'id': 4,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': { 'latitude': 52.35054, 'longitude': 4.908976, 'zoom': 16 },
    'volume': 2,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
    'price': 294,
    'rating': 3.8,
    'title': 'Loft Studio in the Central Area',
    'type': 'hotel',
  }, {
    'bedrooms': 4,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    'goods': [
      'Air conditioning',
      'Baby seat',
      'Fridge',
      'Towels',
      'Breakfast',
      'Washer',
      'Laptop friendly workspace',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 9,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.37854,
      'longitude': 4.894976,
      'zoom': 16,
    },
    'volume': 10,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
    'price': 406,
    'rating': 2.2,
    'title': 'Loft Studio in the Central Area',
    'type': 'hotel',
  }, {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.',
    'goods': [
      'Washer',
      'Towels',
      'Breakfast',
      'Air conditioning',
      'Baby seat',
      'Fridge',
      'Laptop friendly workspace',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 19,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.397540000000006,
      'longitude': 4.9099759999999995,
      'zoom': 16,
    },
    'volume': 2,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
    'price': 286,
    'rating': 2.8,
    'title': 'The Joshua Tree House',
    'type': 'room',
  }, {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    'goods': [
      'Coffee machine',
      'Breakfast',
      'Laptop friendly workspace',
      'Air conditioning',
      'Towels',
      'Baby seat',
      'Dishwasher',
      'Washer',
      'Fridge',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 22,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
    ],
    'isFavorite': false,
    'isPremium': true,
    'location': {
      'latitude': 52.37554,
      'longitude': 4.9019759999999994,
      'zoom': 16,
    },
    'volume': 5,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    'price': 143,
    'rating': 4.6,
    'title': 'Canal View Prinsengracht',
    'type': 'house',
  }, {
    'bedrooms': 2,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.',
    'goods': [
      'Washer',
      'Fridge',
      'Air conditioning',
      'Towels',
      'Baby seat',
      'Breakfast',
      'Laptop friendly workspace',
      'Dishwasher',
      'Coffee machine',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25, 'isPro': true,
      'name': 'Angelina',
    },
    'id': 24,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.367540000000005,
      'longitude': 4.883976,
      'zoom': 16,
    },
    'volume': 9,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    'price': 289,
    'rating': 2,
    'title': 'House in countryside',
    'type': 'house',
  },
];
