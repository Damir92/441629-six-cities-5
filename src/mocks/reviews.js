export const reviews = [
  {
    id: 1,
    avatar: `img/avatar-max.jpg`,
    date: `2019-04-24`,
    dateHuman: `April 2019`,
    name: `Max`,
    rating: 4,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
  {
    id: 2,
    avatar: `img/avatar-max.jpg`,
    date: `2019-04-24`,
    dateHuman: `April 2019`,
    name: `Max`,
    rating: 4,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
];
