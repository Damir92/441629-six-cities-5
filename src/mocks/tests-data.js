export const cityOffers = [
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    goods: [
      `Fridge`,
      `Washer`,
      `Coffee machine`,
      `Washing machine`,
      `Dishwasher`,
      `Laptop friendly workspace`,
      `Cable TV`,
      `Air conditioning`,
      `Baby seat`,
      `Towels`,
      `Breakfast`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37454,
      longitude: 4.881976,
      zoom: 16,
    },
    volume: 1,
    preview: `previewURL`,
    price: 111,
    rating: 3.3,
    title: `Wood and stone place`,
    type: `room`,
  },
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
    goods: [
      `Laptop friendly workspace`,
      `Baby seat`,
      `Towels`,
      `Air conditioning`,
      `Breakfast`,
      `Washer`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 4,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: true,
    isPremium: false,
    location: {
      latitude: 52.35054,
      longitude: 4.908976,
      zoom: 16,
    },
    volume: 2,
    preview: `previewURL`,
    price: 294,
    rating: 3.8,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  }, {
    bedrooms: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a "slow" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    goods: [
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Towels`,
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 9,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37854,
      longitude: 4.894976,
      zoom: 16,
    },
    volume: 10,
    preview: `previewURL`,
    price: 406,
    rating: 2.2,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  },
];
