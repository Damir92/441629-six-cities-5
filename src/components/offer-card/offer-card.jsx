import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {setActiveCard} from '../../store/action';
import {convertRatingToPercent} from '../../utils';

import {LivingType, Routes} from '../../const';
import {OfferCardPropTypes} from '../../prop-types';

const OfferCard = ({offer = {}, onCardEnterMouse, isMainPage = true}) => {

  const {
    id,
    isFavorite,
    isPremium,
    preview,
    price,
    rating,
    title,
    type,
  } = offer;

  const handleMouseEnterCard = () => {
    onCardEnterMouse(offer.id);
  };

  const handleMouseLeaveCard = () => {
    onCardEnterMouse(null);
  };

  return (
    <article
      className={`place-card ${isMainPage ? `cities__place-card` : `near-places__card`}`}
      onMouseEnter={handleMouseEnterCard}
      onMouseLeave={handleMouseLeaveCard}
    >
      {
        isPremium
        &&
        <div className="place-card__mark">
          <span>Premium</span>
        </div>
      }
      <div className={`place-card__image-wrapper ${isMainPage ? `cities__image-wrapper` : `near-places__image-wrapper`}`}>
        <a href="#">
          <img className="place-card__image" src={preview} width="260" height="200" alt="Place image" />
        </a>
      </div>
      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>
          <button
            className={`place-card__bookmark-button button ${isFavorite ? `place-card__bookmark-button--active` : ``}`}
            type="button"
          >
            <svg className="place-card__bookmark-icon" width="18" height="19">
              <use xlinkHref="#icon-bookmark"></use>
            </svg>
            <span className="visually-hidden">
              {isFavorite ? `In bookmarks` : `To bookmarks`}
            </span>
          </button>
        </div>
        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span style={{width: convertRatingToPercent(rating)}}></span>
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <h2 className="place-card__name">

          <Link
            to={`${Routes.OFFER_LINK}/${id}`}
          >
            {title}
          </Link>
        </h2>
        <p className="place-card__type">{LivingType[type]}</p>
      </div>
    </article>
  );
};

OfferCard.propTypes = {
  isMainPage: PropTypes.bool.isRequired,
  offer: PropTypes.shape(OfferCardPropTypes).isRequired,
  onCardEnterMouse: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onCardEnterMouse: (id) => dispatch(setActiveCard(id)),
});

export {OfferCard};
export default connect(null, mapDispatchToProps)(OfferCard);
