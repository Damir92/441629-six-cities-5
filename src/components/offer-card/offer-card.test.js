import React from 'react';
import {shallow} from 'enzyme';

import {PageTypes, AuthorizationStatus} from '../../const';
import {offer, noop} from '../../mocks/tests-data';

import {OfferCard} from './offer-card';

jest.mock(`react-router-dom`, () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
  Link: `Link`,
}));

describe(`OfferCard component`, () => {
  it(`OfferCard main page snapshot`, () => {
    const tree = shallow(
        <OfferCard
          logged={AuthorizationStatus.AUTH}
          offer={offer}
          onCardEnterMouse={noop}
          onFavoriteClick={noop}
          pageType={PageTypes.MAIN}
        />
    );

    expect(tree).toMatchSnapshot();
  });

  it(`OfferCard offer page snapshot`, () => {
    const tree = shallow(
        <OfferCard
          logged={AuthorizationStatus.AUTH}
          offer={offer}
          onCardEnterMouse={noop}
          onFavoriteClick={noop}
          pageType={PageTypes.OFFER}
        />
    );

    expect(tree).toMatchSnapshot();
  });

  it(`OfferCard favorites page snapshot`, () => {
    const tree = shallow(
        <OfferCard
          logged={AuthorizationStatus.AUTH}
          offer={offer}
          onCardEnterMouse={noop}
          onFavoriteClick={noop}
          pageType={PageTypes.FAVORITES}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});

