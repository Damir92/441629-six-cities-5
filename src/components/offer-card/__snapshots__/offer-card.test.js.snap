// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard component OfferCard favorites page snapshot 1`] = `
<article
  className="place-card favorites__card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__image-wrapper favorites__image-wrapper"
  >
    <Link
      to="/offer/9"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={110}
        src="previewURL"
        width={150}
      />
    </Link>
  </div>
  <div
    className="place-card__info favorites__card-info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          406
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button "
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "40%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/9"
      >
        Loft Studio in the Central Area
      </Link>
    </h2>
    <p
      className="place-card__type"
    >
      Hotel
    </p>
  </div>
</article>
`;

exports[`OfferCard component OfferCard main page snapshot 1`] = `
<article
  className="place-card cities__place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__image-wrapper cities__image-wrapper"
  >
    <Link
      to="/offer/9"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="previewURL"
        width={260}
      />
    </Link>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          406
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button "
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "40%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/9"
      >
        Loft Studio in the Central Area
      </Link>
    </h2>
    <p
      className="place-card__type"
    >
      Hotel
    </p>
  </div>
</article>
`;

exports[`OfferCard component OfferCard offer page snapshot 1`] = `
<article
  className="place-card near-places__card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__image-wrapper near-places__image-wrapper"
  >
    <Link
      to="/offer/9"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="previewURL"
        width={260}
      />
    </Link>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          406
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button "
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "40%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/9"
      >
        Loft Studio in the Central Area
      </Link>
    </h2>
    <p
      className="place-card__type"
    >
      Hotel
    </p>
  </div>
</article>
`;
